source/Octopus.Tentacle.Contracts/ScriptTicket.cs

```diff
@@ -18,7 +17,7 @@ public bool Equals(ScriptTicket? other)
                 return false;
             if (ReferenceEquals(this, other))
                 return true;
-            return string.Equals(TaskId, other.TaskId);
+            return string.Equals(TaskId, other.TaskId, StringComparison.OrdinalIgnoreCase);
```

Comment:
This was a remnant of a direction I backed away from.

But I kept it in to start the conversation about whether this is actually a legit fix.

If you look at the dictionaries in `ScriptService`, they both compare using `OrdinalIgnoreCase` on the keys (i.e. script tickets as strings).

I'd say, the script ticket `F2F78486-A53E-4BF9-9939-CD64FB07E37D` is the same as `f2f78486-a53e-4bf9-9939-cd64fb07e37d`.

Reply:
This was made into a [new story.](https://app.shortcut.com/octopusdeploy/story/60772/make-scriptticket-ignore-case-when-doing-comparisons)